### ISSUE TITLE
how to synthesize spherical harmonic coefficients into geoid

### ISSUE THREAD

**Issue by Lydia823** [geodynamics/aspect#1836]
what formula to use to synthesize spherical harmonic coefficients of density anomaly contribution or surface dynamic topography contribution or CMB dynamic topography contribution computed from aspect , parameters set according to manual of page 368-369?

**Comment by gassmoeller** [comment]
@Shangxin-Liu you can probably answer this best. Is there a good paper that contains the equations?

Apart from that Lydia if you are looking for a spatial visualization of the geoid you can use the visualization/geoid postprocessor (as opposed to the normal geoid postprocessor) to output the geoid in the same dataset as the rest of the solution (and visualize with Paraview or however you currently visualize the rest of your data). 
The Geoid::evaluate function in source/postprocess/geoid.cc should also give you a point to start looking for how to convert the coefficients to spatial values.

**Comment by Shangxin-Liu** [comment]
Hi Lydia,

The geoid code output spherical harmonics of density, surface topography, and CMB topography contribution in dimensional form with 4*pi*G/(surface gravity) prefactor. One of documents clearly shows this equation is equation (7) in the paper
http://www.sciencedirect.com/science/article/pii/S0031920113002057

**Comment by Lydia823** [comment]
Dear Shangxin,
Thanks for your advice of the paper, i have downloaded it and will read it carefully.



-----原始邮件-----
发件人: "Shangxin Liu" <notifications@github.com>
发送时间: 2017年7月7日 星期五
收件人: geodynamics/aspect <aspect@noreply.github.com>
抄送: Lydia823 <rhcui@asch.whigg.ac.cn>, Author <author@noreply.github.com>
主题: Re: [geodynamics/aspect] how to synthesize spherical harmonic coefficients into geoid (#1836)



Hi Lydia,

The geoid code output spherical harmonics of density, surface topography, and CMB topography contribution in dimensional form with 4piG/(surface gravity) prefactor. One of documents clearly shows this equation is equation (7) in the paper
http://www.sciencedirect.com/science/article/pii/S0031920113002057

**Comment by ian-r-rose** [comment]
@gassmoeller You are right about the `execute()` function. I think that the synthesis step should still be clearly documented in the manual, since there are an unfortunate number of spherical harmonics normalizations in the literature.

**Comment by Lydia823** [comment]
@gassmoeller Thanks for your good advice, i will look carefully of that in geoid.cc.

**Comment by Shangxin-Liu** [comment]
As we discussed in the hackathon, it would be better to put the spherical harmonic synthesis (back and forth) into utilities.cc because in the future we may need to enable ASPECT to transfer other variables like velocity etc to spherical harmonic domain in the post processing. But we need to choose a universal format for this. I think the current format of normalization is a pretty commonly used one in geophysics. Other details may also need to be taken care of such as the input spherical function's format. @gassmoeller @ian-r-rose

**Comment by ian-r-rose** [comment]
Unfortunately, at least three conventions are commonly used (geodetic, Schmidt, and fully normalized). The one we are using (fully normalized) is the best, I think.

I agree that it should probably be in `utilities.cc`, ultimately.

**Comment by Shangxin-Liu** [comment]
Yeah, the fully normalized one we use is very common in seismology (except S40RTS:-)) 

I personally think this should be the future universal format in ASPECT. @gassmoeller @ian-r-rose

**Comment by gassmoeller** [comment]
I am fine with that, and would support the move. Anyone interested in going forward with a pull request?
